### ISSUE TITLE
mimir query throws `RuntimeException: Boxed Error`

### ISSUE THREAD

**Issue by alissapajer** [precog/quasar#2708]
The query in `demo/timeSeriesAggregation.test`
```sql
SELECT SUM(value) AS measure, timestamp(dt) AS dimension, sensor AS series FROM smalltimeseries GROUP BY sensor, dt ORDER BY sensor ASC, dt ASC
```
throws a `RuntimeException` for mimir. The query passes (on `next-major`) for all other connectors. The `QScript` appears to be correct at first glance, but it's always worth checking again in case the other connectors are masking over a problem.
```
💪 $ SELECT SUM(value) AS measure, timestamp(dt) AS dimension, sensor AS series FROM timeseriesSmall GROUP BY sensor, dt ORDER BY sensor ASC, dt ASC
💪 $ Exception in thread "AeshProcess: 3" java.lang.RuntimeException: Boxed Error
        at quasar.effect.Failure$.quasar$effect$Failure$$$anonfun$16(Failure.scala:112)
        at quasar.effect.Failure$$anon$5.apply(Failure.scala:102)
        at quasar.effect.Failure$$anon$5.apply(Failure.scala:102)
        at scalaz.NaturalTransformation$$anon$1.apply(NaturalTransformation.scala:19)
        at scalaz.NaturalTransformations$$anonfun$natToFunction$1.apply(NaturalTransformation.scala:41)
        at scalaz.$bslash$div.fold(Either.scala:57)
        at quasar.fp.free.package$EnrichNT$$anon$2.apply(package.scala:26)
        at quasar.fp.free.package$EnrichNT$$anon$2.apply(package.scala:26)
        at scalaz.NaturalTransformations$$anonfun$natToFunction$1.apply(NaturalTransformation.scala:41)
        at scalaz.$bslash$div.fold(Either.scala:58)
        at quasar.fp.free.package$EnrichNT$$anon$2.apply(package.scala:26)
        at quasar.fp.free.package$EnrichNT$$anon$2.apply(package.scala:26)
        at scalaz.Free.foldMap(Free.scala:217)
        at scalaz.Free.foldMap(Free.scala:219)
        at scalaz.Free$$anonfun$foldMap$2.apply(Free.scala:219)
        at scalaz.EitherT$$anonfun$flatMap$1$$anonfun$apply$4.apply(EitherT.scala:93)
        at scalaz.$bslash$div.fold(Either.scala:58)
        at scalaz.EitherT$$anonfun$flatMap$1.apply(EitherT.scala:93)
        at scalaz.EitherT$$anonfun$flatMap$1.apply(EitherT.scala:93)
        at scalaz.concurrent.Task$$anonfun$flatMap$1$$anonfun$1.apply(Task.scala:35)
        at scalaz.concurrent.Task$$anonfun$flatMap$1$$anonfun$1.apply(Task.scala:35)
        at scalaz.concurrent.Task$.Try(Task.scala:457)
        at scalaz.concurrent.Task$$anonfun$flatMap$1.apply(Task.scala:35)
        at scalaz.concurrent.Task$$anonfun$flatMap$1.apply(Task.scala:33)
        at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:52)
        at scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:58)
        at scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:58)
        at scalaz.concurrent.Future.step(Future.scala:118)
        at scalaz.concurrent.Future.unsafePerformListen(Future.scala:75)
        at scalaz.concurrent.Future$$anonfun$unsafePerformListen$1$$anonfun$apply$4.apply(Future.scala:79)
        at scalaz.concurrent.Future$$anonfun$unsafePerformListen$1$$anonfun$apply$4.apply(Future.scala:79)
        at scalaz.Free$$anonfun$map$1.apply(Free.scala:99)
        at scalaz.Free$$anonfun$map$1.apply(Free.scala:99)
        at scalaz.Free.resume(Free.scala:117)
        at scalaz.Free.go2$1(Free.scala:161)
        at scalaz.Free.go(Free.scala:165)
        at scalaz.Free.run(Free.scala:271)
        at scalaz.concurrent.Future$$anonfun$async$1$$anonfun$apply$14.apply(Future.scala:428)
        at scalaz.concurrent.Future$$anonfun$async$1$$anonfun$apply$14.apply(Future.scala:428)
        at delorean.package$FutureAPI$$anonfun$toTask$1$$anonfun$apply$3$$anonfun$apply$2.apply$mcV$sp(package.scala:32)
        at delorean.package$FutureAPI$$anonfun$toTask$1$$anonfun$apply$3$$anonfun$apply$2.apply(package.scala:32)
        at delorean.package$FutureAPI$$anonfun$toTask$1$$anonfun$apply$3$$anonfun$apply$2.apply(package.scala:32)
        at scalaz.concurrent.StrategysLow$$anon$3$$anon$4.call(Strategy.scala:77)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by alissapajer** [comment]
@djspiewak @edmundnoble I've narrowed it down to this query, which also throws the same exception
```sql
SELECT timestamp(dt) AS dimension FROM timeseriesSmall 
```
so I think it's a problem with `timestamp`.

**Comment by edmundnoble** [comment]
Looks like. https://github.com/quasar-analytics/quasar/blob/ba9499e8b1302aa51f7caf45565cf5f846bc4fb4/mimir/src/main/scala/quasar/mimir/MapFuncCorePlanner.scala#L96

**Comment by alissapajer** [comment]
Oh, yeah that would explain it. I wonder why the unimplemented error is propagating out as a `Boxed Error`.

**Comment by edmundnoble** [comment]
@djspiewak is right to assign this to me, it'll be fixed as part of #2564.

**Comment by zeeiyerWork** [comment]
since this will be fixed when the time stuff merges, moving this out.
